File: Leetcode-Problems/Maximum Ascending Subarray Sum.cpp
class Solution {
public:
    int maxAscendingSum(vector<int>& nums) {
        int sum = nums[0];
        int ans = nums[0];
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] > nums[i - 1]) {
                sum += nums[i];
            } else {
                ans = max(ans, sum);
                sum = nums[i];
            }
        }
        return max(ans, sum);
    }
};

File: Leetcode-Problems/Maximum Ascending Subarray Sum.java
class Solution {
    public int maxAscendingSum(int[] nums) {
        int ans = 0, temp = nums[0];
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] <= nums[i - 1]) {
                ans = Math.max(ans, temp);
                temp = 0;
            }
            temp += nums[i];
        }
        ans = Math.max(ans, temp);
        return ans;
    }
}